src/jit/codegencommon.cpp

```diff
@@ -10611,6 +10612,17 @@ void CodeGen::genPreserveCalleeSavedFltRegs(unsigned lclFrameSize)
         regMaskTP regBit = genRegMask(reg);
         if ((regBit & regMask) != 0)
         {
+#ifdef FEATURE_AVX_SUPPORT
+            // when we reach here, function does not contain AVX instruction so far, however, since copyIns can
+            // be an AVX instruction such as vmovupd, we should check and issue vzeroupper before the copyIns to
+            // avoid Legacy SSE code (from native code such as Reverse PInvoke) to AVX transition penalty
+            if (compiler->getFloatingPointInstructionSet() == InstructionSet_AVX && !bVzeroupperIssued &&
+                getEmitter()->IsAVXInstruction(copyIns))
+            {
+                instGen(INS_vzeroupper);
+                bVzeroupperIssued = true;
+            }
+#endif
```

Comment:
Windows xarch has callee saved float regs whereas Unix x64 no callee saved regs.

On Windows, we would be saving callee saved float regs only if method has floating point or SIMD codegen.  In such a case lowerxarch.cpp logic will set the required AVX flags on emitter.  In such a case why do we need this logic here?

In other words, what is the case in which AVX flags are not set but method is attempting to save callee saved float regs to stack in prolog?